File: 0231-power-of-two/0231-power-of-two.py
class Solution:
    def isPowerOfTwo(self, n: int) -> bool:
        
        # Check if n is equal to 1 or n divided by 2 is equal to 1
        # This is to handle the case when n is a power of two (e.g., 1, 2, 4, 8, ...)
        if n / 2 == 1 or n == 1:
            return True

        # If n is greater than 2 and divisible by 2,
        # recursively call isPowerOfTwo with n/2
        if n > 2:
            return self.isPowerOfTwo(n/2)
        else:
            # If n is not a power of two, return False
            return False

File: 0231-power-of-two/0231-power-of-two.java
class Solution {
    public boolean isPowerOfTwo(int n) {

        // to return False for negative values
        if ( n <= 0 ){
            return false;
        }

        while( n%2 == 0 ){
            n = n/2;
        }

        if ( n == 1 ){
            return true;
        }
        else{
            return false;
        }

        
    }
}